src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,61 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.producer.BulldozerProducer;
+import core.mate.academy.producer.ExcavatorProducer;
+import core.mate.academy.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList();
+        final String typeSimpleName = type.getSimpleName();
+        switch (typeSimpleName) {
+            case "Truck":
```

Comment:
do switch of if-else on type itself (compare with `Bulldozer.class` etc), no need to get `simpleName`